lib/facter/jenkins.rb

```diff
@@ -2,10 +2,13 @@
 #
 # Creates a fact 'jenkins_plugins' containing a comma-delimited string of all
 # jenkins plugins + versions.
-#
-#
-require 'puppet/jenkins/facts'
+require 'facter'
+require_relative '../puppet/jenkins/plugins'
 
-# If we're being loaded inside the module, we'll need to go ahead and add our
-# facts then won't we?
-Puppet::Jenkins::Facts.install
+Facter.add(:jenkins_plugins) do
+  confine kernel: 'Linux'
+  setcode do
+    plugins = Puppet::Jenkins::Plugins.available
+    plugins.keys.sort.map { |plugin| "#{plugin} #{plugins[plugin][:plugin_version]}" }.join(', ')
```

Comment:
it would be so nice and easy to provide a structured fact :(

Reply:
True and it would simplify other things. IMHO we should have a `jenkins` fact that's roughly: `{jenkins: {plugins: {name1: version1, name2: version2}}}`. However, let's keep that for a later PR.

Reply:
I totally agree. I'm fine with merging this as is

Reply:
I started this before the big modulesync PR update. I'd prefer to rebase on that once it's merged.